src/source/src/manager.rs

```diff
@@ -215,7 +215,7 @@ impl SourceManager for MemSourceManager {
             columns: source_columns,
             format: SourceFormat::Invalid,
             row_id_index: 0, // always use the first column as row_id
-            row_id_generator: Arc::new(Mutex::new(RowIdGenerator::with_epoch(
```

Comment:
So `std::sync::Mutex` does not support deterministic tests? 👀

Reply:
`std::sync::MutexGuard` can not cross any `.await` point, or it will make the future `!Send`. 😇

Reply:
Where do we require `Send` before?